src/prefect_collection_registry/utils.py

```diff
@@ -0,0 +1,454 @@
+import asyncio
+import base64
+import inspect
+import json
+import os
+import tempfile
+from collections.abc import Callable, Generator
+from multiprocessing import Lock
+from pkgutil import iter_modules
+from types import ModuleType
+from typing import Any, Literal
+
+import fastjsonschema
+import httpx
+import yaml
+from gh_util.client import GHClient
+from gh_util.types import GitHubRepo
+from prefect import Flow, task
+from prefect.blocks.system import Secret
+from prefect.utilities.importtools import load_module, to_qualified_name
+from pydantic_core import from_json
+
+CollectionViewVariety = Literal["block", "flow", "worker"]
+
+# Create a lock file in a location accessible to all processes
+LOCK_FILE = os.path.join(tempfile.gettempdir(), "prefect_collection_registry.lock")
+_file_lock = Lock()
+
+
+def skip_parsing(
+    name: str, obj: ModuleType | Callable[..., Any], module_nesting: str
+) -> bool:
+    """Skips parsing the object if it's a private object or if it's not in the
+    module nesting, preventing imports from other libraries from being added to the
+    examples catalog.
+    """
+    try:
+        wrong_module = not to_qualified_name(obj).startswith(module_nesting)
+    except AttributeError:
+        wrong_module = False
+    return obj.__doc__ is None or name.startswith("_") or wrong_module
+
+
+def find_flows_in_module(
+    module_name: str,
+) -> Generator[Flow[..., Any], None, None]:
+    """Finds all flows in a module."""
+    module = load_module(module_name)
+
+    for _, name, ispkg in iter_modules(module.__path__):
+        if ispkg:
+            # catch submodules that are not top-level
+            # e.g. prefect-hightouch.syncs.flows
+            yield from find_flows_in_module(f"{module_name}.{name}")
+        else:
+            try:
+                submodule = load_module(f"{module_name}.{name}")
+                print(f"\tLoaded submodule {module_name}.{name}...")
+            except ModuleNotFoundError:
+                continue
+
+            for _, obj in inspect.getmembers(submodule):
+                if skip_parsing(name, obj, module_name):
+                    continue
+                if isinstance(obj, Flow):
+                    yield obj
+
+
+async def get_file_contents(
```

Comment:
What's the difference between this and `get_file_content` and `get_repo_contents`?

Reply:
ill deduplicate